packet/packet.py

```diff
@@ -43,38 +45,101 @@ def sign(signer_username, freshman_username):
+    if eboard is not None:
+        packet.info_eboard = eboard
+    if events is not None:
+        packet.info_events = events
+    if achieve is not None:
+        packet.info_achieve = achieve
+    try:
+        db.session.commit()
+    except exc.SQLAlchemyError:
+        return False
+    return True
+
+
 @lru_cache(maxsize=2048)
 def get_signatures(freshman_username):
+    """
+    Gets a list of all signatures for the given member
+    :param freshman_username:
+    :return:
+    """
     packet = Freshman.query.filter_by(rit_username=freshman_username).first().current_packet()
-    eboard = UpperSignature.query.filter_by(packet_id=packet.id, eboard=True).order_by(UpperSignature.signed.desc())
-    upper_signatures = UpperSignature.query.filter_by(packet_id=packet.id, eboard=False).order_by(
-        UpperSignature.signed.desc())
-    fresh_signatures = FreshSignature.query.filter_by(packet_id=packet.id).order_by(FreshSignature.signed.desc())
-    misc_signatures = MiscSignature.query.filter_by(packet_id=packet.id)
+
+    eboard = db.session.query(UpperSignature.member, UpperSignature.signed, Freshman.rit_username) \
+        .select_from(UpperSignature).join(Packet).join(Freshman) \
+        .filter(UpperSignature.packet_id == packet.id, UpperSignature.eboard.is_(True)) \
+        .order_by(UpperSignature.signed.desc()) \
+        .distinct().all()
+
+    upper_signatures = db.session.query(UpperSignature.member, UpperSignature.signed, Freshman.rit_username) \
+        .select_from(UpperSignature).join(Packet).join(Freshman) \
+        .filter(UpperSignature.packet_id == packet.id, UpperSignature.eboard.is_(False))\
+        .order_by(UpperSignature.signed.desc())\
+        .distinct().all()
+    fresh_signatures = \
+    db.session.query(FreshSignature.freshman_username, FreshSignature.signed, Freshman.rit_username, Freshman.name) \
+        .select_from(Packet).join(FreshSignature).join(Freshman) \
+        .filter(FreshSignature.packet_id == packet.id) \
+        .order_by(FreshSignature.signed.desc()) \
+        .distinct().all()
+
+    misc_signatures = db.session.query(MiscSignature.member, Freshman.rit_username)\
+        .select_from(MiscSignature).join(Packet).join(Freshman) \
+        .filter(MiscSignature.packet_id == packet.id) \
+        .distinct().all()
+
     return {'eboard': eboard,
             'upperclassmen': upper_signatures,
             'freshmen': fresh_signatures,
             'misc': misc_signatures}
 
 
+def get_misc_signatures():
+    packet_misc_sigs = {}
+    try:
+        result = db.engine.execute("SELECT packet.freshman_username "
+                                   "AS username, count(signature_misc.member) "
+                                   "AS signatures FROM packet "
+                                   "RIGHT OUTER JOIN signature_misc "
+                                   "ON packet.id = signature_misc.packet_id "
+                                   "GROUP BY packet.freshman_username;")
```

Comment:
I'm fairly certain this can be done without raw SQL queries

Reply:
If you do leave this as raw SQL please using triple quote strings.